src/Illuminate/Filesystem/Filesystem.php

```diff
@@ -331,6 +332,41 @@ public function move($path, $target)
         return rename($path, $target);
     }
 
+    /**
+     * Create a file from a base64 image
+     *
+     * @param  string  $base64File
+     * @return string
+     */
+    public function fromBase64(string $base64File)
+    {
+        if (!$this->isBase64File($base64File)) {
+            throw new InvalidBase64FileException();
+        }
+
+        if (str_contains($base64File, 'data:image')) {
```

Comment:
Are you sure, this only applies to data URIs for `image/*` MIME types but not other kinds of files with the same data URI segments?

Reply:
I started this with only images in mind and while i was on it I remembered midway that it should be open to other kinds of files.

Reply:
So i started writing the function with images in mind then i realized midway "What about other functions?". Thanks for pointing that out

Reply:
Yeah, that's probably one of the most frequent use cases. Maybe that's why.